### DISCUSSION TITLE
API docs arent detailed enough

### DISCUSSION THREAD

**Discussion by abdulrahmanalaa123** [pmndrs/zustand#2327] [General] [answered]
I've read the get started im new to js but I cant seem to find any api doc like for example the set() API I tried finding any detailed API doc for it like its options,etc. couldnt find any 
referenced from your get started:
"The set function has a second argument, false by default. Instead of merging, it will replace the state model. Be careful not to wipe out parts you rely on, like actions."
The grammar is so misleading, the merging behavior what does it reference? is it true or the false setting?
Now i dont like using chatgpt and reading the API docs usually does it for me but asking what the set options are there are 4 options which i wouldve never known from the current api docs.
"
state: A function that receives the current state of the store and returns the new state. This function is called with the current state as its argument and should return the updated state.
options: An object containing options to control the behavior of the state update. The available options are:

name: A string specifying the name of the action. This is useful for debugging and logging purposes.

immediate: A boolean indicating whether to notify subscribers immediately after the state update. When set to true, subscribers are notified synchronously. Default is false.

state: A boolean indicating whether to merge the new state with the current state or replace the current state entirely. When set to true, the new state is merged with the current state. Default is false (replace).

persist: A boolean indicating whether to persist the state update to any configured persistence middleware. Default is false.
 "
 I've tried immediate which I needed and it worked so it supposedly wasnt nonesense gibberish now i see that this is similar to https://github.com/pmndrs/zustand/issues/206 but i feel that the issue hasnt been adressed there well maybe i didnt look hard enough for the api docs but i felt that posting an issue would be good.
P.S : I'm not experienced by any means in either react or js but i felt like ive tried hard enough for me to post an issue here for the api docs

**Comment by dbritto-dev** [comment]
@abdulrahmanalaa123 we're working on that if you have some suggestions please let us know your thoughts

**Comment by abdulrahmanalaa123** [comment]
@dbritto-dev  So I've faced alot of difficulties with differentiating between subscribing and using normal hook selectors the shallow property example code isnt that great since its a hook and none of the variables are inside a component which confused me a bit as well as the files are examples which wouldnt help me where im looking for something since i dont know what exactly am i looking for my general route for learning something is usually glancing at most of its functionality understanding what i can and then think about the thing that im actually doing so for me that wasnt really helpful as for certain improvements i cant really tell you guys since im not experienced i just shared my thoughts about certain aspects which i roughly understood since i didnt look at the testing and using redux dev tools part since i dont need it atm i would love to be a part of this and help but i dont believe i could actually impact that much other than sharing my opinion.

**Comment by dbritto-dev** [comment, marked as answer]
@abdulrahmanalaa123 sure, let me explain you you a little bit.

We can use Zustand w/o react. 

If you want to use Zustand with vanilla JavaScript (no react at all), you can import `createStore` from `zustand/vanilla`. 

If you want to use Zustand with react, you can import `create` from `zustand`.

There's no difference between them  except that you can't use a vanilla store on react without `useStore` or `useStoreWithEqualityFn` 

create/createStore arguments: 
- set: method that we can use to update the state, set can be use passing a value directly or with a callback and the callback argument is the state itself. Let's say that works exactly like a pure set state on react but shallow merge the current state with the new values that you are trying to setting up (talking about objects/dicts). There's a second parameter that we can use to replace the whole state with a new one instead of shallow merge the new values with the current state.
Some examples: `set({})`, `set(state => ({}))`, `set({}, true)`
- get: method we can use to get the state
- store: the store itself (you don't need this but often use by lib creators)
